lib/checkfunctions.cpp

```diff
@@ -216,9 +216,17 @@ void CheckFunctions::checkIgnoredReturnValue()
             }
 
             if ((!tok->function() || !Token::Match(tok->function()->retDef, "void %name%")) &&
-                (mSettings->library.isUseRetVal(tok) || (tok->function() && tok->function()->isAttributeNodiscard())) &&
+                ((mSettings->library.getUseRetValType(tok) != Library::UseRetValType::NONE) ||
```

Comment:
I have the feeling that it's redundant to check

    ((mSettings->library.getUseRetValType(tok) != Library::UseRetValType::NONE) ||
     (tok->function() && tok->function()->isAttributeNodiscard()))

Because those checks are more or less repeated ..